### ISSUE TITLE
Redundant Space at top of Bar Chart

### ISSUE THREAD

**Issue by verofy-simon-denny** [Abhinandan-Kushwaha/react-native-gifted-charts#71]
Hi,

Just wondering if it's possible to get rid of the redundant space at the top of the bar chart - highlighted the background in red to show the space at the top:

<img width="288" alt="Screenshot 2022-01-20 at 17 10 34" src="https://user-images.githubusercontent.com/19588979/150388714-3a13d3cd-537b-437a-b25a-b3a947f8a415.png">

Thanks

.

**Comment by verofy-simon-denny** [comment]
Sorry, also, is there a way that we can set the space on the left of the chart to 0 for times when we don't want the y axis showing?

**Comment by Abhinandan-Kushwaha** [comment]
Hi @verofy-simon-denny 👋

Regarding your [first question](https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/issues/71#issue-1109564153) -

Yes, we can remove the extra padding at the top, by using the `maxValue` prop.

Here's a chart without using the `maxValue` prop-
<img width="211" alt="padding" src="https://user-images.githubusercontent.com/20596944/150563321-a0d6e9b4-cb13-4f3d-afa5-80ef97f163e1.png">
 The code for the above chart is -
```js
const barData = [{value: 15}, {value: 30}, {value: 26}, {value: 40}];
<BarChart backgroundColor={'red'} data={barData} />
```

Now, if we add `maxValue={40}` -
```js
const barData = [{value: 15}, {value: 30}, {value: 26}, {value: 40}];
<BarChart backgroundColor={'red'} maxValue={40} data={barData} />
```
The output becomes-

<img width="206" alt="maxValue" src="https://user-images.githubusercontent.com/20596944/150567868-e65f55c1-7085-49da-95b9-e40a779c9bef.png">
Notice the padding at the top is now gone!

How it works-

If we don't provide the `maxValue`, it computes the maxValue as follows-
It takes the biggest value in the data array and the next reading on the y axis becomes the maxValue.
For example, in the above case, the biggest value in the data array is `40` so the next reading on y axis, i.e. `50` becomes the maxValue.

**Comment by verofy-simon-denny** [comment]
Thank you for the answers, appreciate the time and the great library.

It makes sense, but I can't seem to set the height and max value at the same time, am I doing something wrong here?

I've used your example, but added height:

`<BarChart
            backgroundColor={'red'}
            maxValue={40}
            data={barData}
            height={140}
          />`

<img width="275" alt="Screenshot 2022-01-21 at 17 43 54" src="https://user-images.githubusercontent.com/19588979/150575054-8714327c-be49-485b-a250-8a8febc10ede.png">

Many Thanks, Simon.

**Comment by Abhinandan-Kushwaha** [comment]
Yes @verofy-simon-denny, Even I faced the same issue. But the fact is there's no issue as such. I reloaded the app and it worked like a charm💫✨

Please try reloading the app after changing any of these props- `maxValue`, `stepValue`, `stepHeight`, `height`
If it's not solved even after reloading, please do let me know, I will work on it.

**Comment by verofy-simon-denny** [comment]
Thanks that works now.

**Comment by Hanuman-Ghawade** [comment]
> Thank you for the answers, appreciate the time and the great library.
> 
> It makes sense, but I can't seem to set the height and max value at the same time, am I doing something wrong here?
> 
> I've used your example, but added height:
> 
> `<BarChart backgroundColor={'red'} maxValue={40} data={barData} height={140} />`
> 
> <img alt="Screenshot 2022-01-21 at 17 43 54" width="275" src="https://user-images.githubusercontent.com/19588979/150575054-8714327c-be49-485b-a250-8a8febc10ede.png">
> 
> Many Thanks, Simon.

Hi @verofy-simon-denny ,

I am facing the same issue. But it only happens when I use the topLabelComponent. Otherwise, it is working fine.

Please suggest any answer.

Thanks in advance.
